pkg/controller/datavolume-controller.go

```diff
@@ -1555,8 +1555,9 @@ func (r *DatavolumeReconciler) expand(log logr.Logger,
 	if !podExists {
 		var err error
 		pod, err = r.createExpansionPod(pvc, dv, podName)
-		if err != nil {
-			return false, handleFailedPod(err, podName, pvc, r.recorder, r.client)
+		// Check if pod has failed and, in that case, record an event with the error
+		if podErr := handleFailedPod(err, podName, pvc, r.recorder, r.client); podErr != nil {
```

Comment:
I actually liked it better like the first way you did it but style preferences :)

Reply:
hmmm I'm torn between the two. I think the current one is easier to read but maybe too verbose.

Reply:
you dont need to change again thats ok